src/Generator/ClassGenerator.php

```diff
@@ -430,6 +430,23 @@ public function getExtendedClass()
     }
 
     /**
+     * @return boolean
+     */
+    public function hasExtentedClass()
+    {
+        return !empty($this->extendedClass);
```

Comment:
if `empty()` is used to compare against `null`, then use `null !== $this->extendedClass` instead, as an empty string would pass this check (and would be invalid)